### PR TITLE
v1.1.1 done

### PR DESCRIPTION
**Pull request by kamiltrzmiel**
Description of the startup steps:

1. Install node.js modules - npm install;
2. Run server.js - default port 3000 - 127.0.0.1:3000 in URL bar;
3. GET all contact list from MongoDB - http://127.0.0.1:3000/api/contacts/
4. GET contact by id from MongoDB - http://127.0.0.1:3000/api/contacts/__id__
5. POST - create new contatct with _id from MongoDB - http://127.0.0.1:3000/api/contacts/ + create raw json in body
6. PUT - update contact from DB - http://127.0.0.1:3000/api/contacts/__id__ + update raw json in body
7. DELETE - delete contact from DB - http://127.0.0.1:3000/api/contacts/__id__
8. PATCH - update 'favorite' in contact from DB - http://127.0.0.1:3000/api/contacts/__id__ + update raw json in body

### CODE REVIEW

File: routes/api/contacts.js [hunk 1]
```diff
@@ -1,84 +1,88 @@
...
   getContactById,
   removeContact,
   addContact,
   updateContact,
+  updatedStatusContact,
 } from './../../models/contacts.js';
 
 export const router = express.Router();
 
-const contactSchema = Joi.object({
```
Comment (AdamChodowiecGoIt91) [inline routes/api/contacts.js:14]: why did you remove Joi validation? :)

File: routes/api/contacts.js [hunk 2]
```diff
@@ -1,84 +1,88 @@
...
-    res.json({ message: 'Contact by id', response });
+    return res.json({ message: 'Contact by id', response });
   } catch (error) {
+    res.status(500).json('error - db get contact by id');
     next(error);
   }
 });
 
 router.post('/', async (req, res, next) => {
+  const { body } = req;
```
Comment (AdamChodowiecGoIt91) [inline routes/api/contacts.js]: please add `Joi` validation for all `post` `put` `patch` endpoints

File: routes/api/contacts.js [hunk 3]
```diff
@@ -1,84 +1,88 @@
...
     next(error);
   }
 });
+
+router.patch('/:contactId', async (req, res, next) => {
+  const { contactId } = req.params;
+  const { body } = req;
+  const { favorite } = body;
+
+  if (!('favorite' in body) || Object.keys(body).lenght === 0) {
```
Comment (AdamChodowiecGoIt91) [inline routes/api/contacts.js]: it can be simplified

File: routes/api/contacts.js (hunk 1 above)
Comment (kamiltrzmiel) [reply routes/api/contacts.js:14]: added in v 1.1.0 :)

File: routes/api/contacts.js (hunk 2 above)
Comment (kamiltrzmiel) [reply routes/api/contacts.js]: added in v 1.1.0 :)

File: controllers/controllers.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  res.status(201).json({ message: 'Contact added', response });
+};
+
+export const removeContact = async (req, res) => {
+  const { contactId } = req.params;
+  const response = await Contact.findByIdAndRemove(contactId);
+  if (!response) {
+    throw errorRequest(404, 'Not found');
+  }
+  res.json({
```
Comment (AdamChodowiecGoIt91) [inline controllers/controllers.js]: add status even if you know that default is `200` :)

File: server.js [hunk 5]
```diff
@@ -6,12 +6,24 @@ config();
...
+  }, 80);
+};
+
 const runServer = async () => {
   try {
     const connection = await mongoose.connect(uriDb);
     console.log('Database connection successful');
     app.listen(3000, () => {
-      console.log('Server running. Use our API on port: 3000');
+      console.log('Host: 127.0.0.1:3000/api/contacts/');
```
Comment (AdamChodowiecGoIt91) [inline server.js]: To nie prawda. Ten wcześniejszy log który usunąłeś był bardziej poprawny ;)

File: controllers/controllers.js (hunk 4 above)
Comment (kamiltrzmiel) [reply controllers/controllers.js]: Ohh ok, 😉

File: server.js (hunk 5 above)
Comment (kamiltrzmiel) [reply server.js]: Przywrócę go Asap jutro 😜

**Review by kamiltrzmiel — COMMENTED** [review]
Done
